File: gin_player/strategy/strategy.go
package strategy

func Fibonacci(n uint64) uint64 {
	switch n {
	case 0:
		return 0
	case 1:
		return 1
	}
	return Fibonacci(n-1) + Fibonacci(n-2)
}

func FibonacciIterative(n uint64) uint64 {
	switch n {
	case 0:
		return 0
	case 1:
		return 1
	}

	var prev, curr uint64 = 0, 1
	for i := uint64(2); i <= n; i++ {
		next := prev + curr
		prev = curr
		curr = next
	}
	return curr
}

func BurnCPU(iterations int) float64 {
	x := 0.5
	r := 3.9999999
	for i := 0; i < iterations; i++ {
		x = r * x * (1.0 - x)
	}
	return x
}

File: axum_player/src/game/strategy.rs
pub fn fibonacci(n: u128) -> u128 {
    if n == 0 {
        0
    } else if n == 1 {
        1
    } else {
        fibonacci(n - 1) + fibonacci(n - 2)
    }
}

pub fn fibonacci_iterative(n: u128) -> u128 {
    if n == 0 {
        return 0;
    } else if n == 1 {
        return 1;
    }

    let mut prev: u128 = 0;
    let mut curr: u128 = 1;

    for _ in 2..=n {
        let next = prev.checked_add(curr).unwrap_or(curr);
        prev = curr;
        curr = next;
    }

    curr
}

pub fn burn_cpu(iterations: usize) -> f64 {
    let mut x = 0.5f64;
    let r = 3.9999999f64;
    for _ in 0..iterations {
        x = r * x * (1.0 - x);
    }
    x
}
